wandb/sdk/wandb_artifacts.py

```diff
@@ -155,7 +156,7 @@ class Artifact(ArtifactInterface):
     def __init__(
         self,
         name: str,
-        type: str,
+        type: str = "dataset",
```

Comment:
seems redundant, see a different commit for more details :)

Reply:
We had a discussion on using a default "Undefined" type rather than default dataset